src/nvim/message.c

```diff
@@ -2438,7 +2438,7 @@ static void redir_write(char_u *str, int maxlen)
     if (*s != '\n' && *s != '\r') {
       while (cur_col < msg_col) {
         if (redir_reg)
-          write_reg_contents(redir_reg, (char_u *)" ", -1, TRUE);
+          write_reg_contents(redir_reg, (char_u *)" ", STRLEN(" "), true);
```

Comment:
noted for the strlen, I think I also prefer no macro if not necessary. I will fix this also.